### PR TITLE
Add Finnish localization

### LINKED ISSUE #424: Finnish localisation of onboarding views

### PR DESCRIPTION
**Pull request by tillprochaska**
**🚨 Please do not merge this yet. While most translations are ready, some are still missing. Also, I’m waiting for Isa to approve the changes.**

These changes are intended to be temporary, i.e. we need a multi-lang onboarding page for the #Schweinesystem project, but it’s rather unlikely that we have to support a multi-lang project in the near-future.

It’s important to merge these changes before #440, #434, and #437, so that we can deploy the translations for the #Schweinesystem to complete the onboarding of the Finnish participants.

**Missing translations:**
 * [x] In der Kopfzeile: eine Dialog-Recherche von tactile.news
 * [x] Titel nach erfolgreicher Anmeldung: Danke für Ihre Anmeldung zu #Schweinesystem!
 * [x] Text nach erfolgreicher Anmeldung: Unsere Dialog-Recherche startet bald. Wir melden uns dann wieder bei Ihnen.
 * [x] Die Willkommens-Nachricht, die bei Telegram versandt wird:
Herzlich Willkommen bei #Schweinesystem. Danke, dass Sie an unserer Dialog-Recherche teilnehmen. In den kommenden Wochen nutzen wir Telegram für den Austausch. Ein Bot arbeitet wie ein Postbote für uns und stellt Fragen und Antworten zu. Ihre Nachrichten lesen und beantworten wir aber selbstverständlich persönlich. Wir freuen uns sehr auf den Dialog mit Ihnen!
Katharina Jakob, Jens Eber, Oliver Eberhardt, Isabelle Buckow, Astrid Csuraji und Jakob Vicari
 * [x] Die Fehler-Nachricht, wenn jemand bei Telegram nicht unterstützte Inhalte sendet:
Unsere Software kann derzeit Text- und Sprachnachrichten und Bilder verarbeiten, jedoch noch keine weiteren Inhalte wie z.B. Dateianhänge. Für die Zwischenzeit bitten wir dich, Nachrichten mit solchen Inhalten an @FrauCsu (Astrid Csuraji) zu schicken. Vielen Dank!
 * [x] In der Fußzeile: Impressum

Close #424 

localhost_3000_onboarding_token=abc123 locale=fi(iPad) (3)

### CODE REVIEW

**Comment by tillprochaska** [comment]
@roschaefer @phoeinx @mattwr18 Would be great if someone could have a look at these changes. When your happy, please don’t merge though, as I’m still waiting for a few missing translations.

**Comment by FrauCsu** [comment]
What changes need to be approved by Isa @tillprochaska ? Please keep me in the loop to speed up things

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -13,4 +14,19 @@ def authenticate
       password: ENV['BASIC_AUTH_LOGIN_PASSWORD']
     )
   end
+
+  def use_locale(&action)
+    locale = locale_params[:locale] || I18n.default_locale
```
Comment (roschaefer) [inline app/controllers/application_controller.rb:19]: Just one thing as I'm curious: How does the rails understand `locale_params[:locale]`? I'm used to set that in the browser, based on `navigator.language` among other things. Does Rails do some magic with `Accept-Language` header? Or do you send the locale in the query string: `?locale=fi`?

File: config/locales/fi.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+fi:
```
Comment (roschaefer) [inline config/locales/fi.yml:1]: Well, I can't speak Finnish, so I cannot really review this :slightly_smiling_face: I would use DeepL or Google Translate to sanity-check the translations.

**Review by roschaefer — APPROVED** [review]
finnish

Looks good to me!

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (tillprochaska) [reply app/controllers/application_controller.rb:19]: For this use case, I opted to set them using a query param `locale=fi`, yes. This makes it easy to send the link to the Finnish translation to Finnish participants and the link for German translation to German participants. In this use case, I think that’s simply the most reliable and easy way, although for most other use cases, using the `Accept-Language` header would probably be better.

File: config/locales/fi.yml (hunk 2 above)
Comment (tillprochaska) [reply config/locales/fi.yml:1]: Did that using Google Translate and also sent a screenshot of the translated page to Isa to have a look at! :)

**Comment by tillprochaska** [comment]
> What changes need to be approved by Isa @tillprochaska ? Please keep me in the loop to speed up things

Nothing important, I just sent her screenshots of the translated page so she could check all the translations are in the correct place while I’m waiting for the remaining translations.

**Comment by tillprochaska** [comment]
@FrauCsu @drjakob: As we don’t know which language someone speaks when they sign up Telegram, I’ve now updated our translations to always send welcome and error message in both, German and Finnish. Hope that’s fine.

IMG_B1FA32FEB2F8-1

File: config/locales/de.yml [hunk 3]
```diff
@@ -225,10 +227,14 @@ de:
...
+
+      Ohjelmistomme pystyy tällä hetkellä käsittelemään teksti- ja ääniviestejä ja kuvia, mutta ei muita sisältöjä, kuten esim. tiedostoliitteitä. Voisitteko ystävällisesti lähettää viestit jossa on liite osoitteeseen @FrauCsu (Astrid Csuraji). Paljon kiitoksia!
     welcome_message: |
-      Herzlich Willkommen bei %{project_name}. Danke, dass Sie an unserer Dialog-Recherche teilnehmen. In den kommenden Wochen nutzen wir Telegram für den Austausch. Ein Bot arbeitet wie ein Postbote für uns und stellt Fragen und Antworten zu. Ihre Nachrichten lesen und beantworten wir aber selbstverständlich persönlich. Wir freuen uns sehr auf den Dialog mit Ihnen!
+      Herzlich Willkommen beim #Schweinesystem. Danke, dass Sie an unserer Dialog-Recherche teilnehmen. In den kommenden Wochen nutzen wir Telegram für den Austausch. Ein Bot arbeitet wie ein Postbote für uns und stellt Fragen und Antworten zu. Ihre Nachrichten lesen und beantworten wir aber selbstverständlich persönlich. Wir freuen uns sehr auf den Dialog mit Ihnen!
```
Comment (tillprochaska) [inline config/locales/de.yml:233]: @roschaefer I removed the ability to set the project name via env variables for this translation, as it doesn’t work for this particular multi-lang project. I guess that’s okay, as it’s soon going to be outdated anyways when we merge #434?

Comment (roschaefer) [reply config/locales/de.yml:233]: That's OK

Curious: why does it not work?

Comment (tillprochaska) [reply config/locales/de.yml:233]: It doesn’t work was maybe not accurate. The problem is that we need to translate the project name as well in this case.

**Pull request merged**